### PR TITLE
feat: add github actions to auto-publish to npmjs

### PR DESCRIPTION
**Pull request by meadowsjared**
This PR introduces a new GitHub Actions workflow (.github/workflows/ci.yaml) to automate our CI/CD process. The workflow is designed to:

- Automatically trigger builds when changes are pushed to the main branch.
- Publish the built package to npmjs upon a successful build.

Key Changes you might want to tweak:
- I included a build step at the end to automatically bump the version number if the pushing is successful
you may want to remove this step if you don't like it
however, I find it helpful since npmjs requires the number to change with each release

To remove the "Automated Version Bump", just remove lines [43-45](https://github.com/QarthO/qHotkeys/pull/6/commits/0c71f8038c97af96629d472d8a19895a49b7e7e3#r1685764599) in `ci.yaml`

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        node-version: '20'
+        registry-url: 'https://registry.npmjs.org'
+    - name: Publish to npm
+      id: publish  # Assign an ID to this step
+      run: npm publish --access public
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    - name: Automated Version Bump
+      if: steps.publish.outcome == 'success'
+      uses: phips28/gh-action-bump-version@v11.0.7
```
Comment (meadowsjared) [inline .github/workflows/ci.yaml:56]: If you don't want the action to change your version number with each release, feel free to delete these lines

**Comment by meadowsjared** [comment]
Also of note: in order to set it up to be able to publish to npmjs.com, you need to do 2 simple steps:
1. go to npmjs.com and generate an access token:
image
note: it'll look something like this:
npm_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX

2, go into https://github.com/QarthO/qHotkeys/settings and set a secret for the repo:
image

3. under repository secrets, add the token you created in step 1, and make sure to name it
`NPM_TOKEN`
image

Once you've done all this, your github action should automatically publish whenever your push to main 🪄

**Comment by meadowsjared** [comment]
@QarthO it should be noted, that if you set this up, this will also happen to resolve issue https://github.com/QarthO/qHotkeys/issues/5
It will also prevent it from happening in the future, as well as making releases less of a headache 👍

**Comment by QarthO** [comment]
I love the idea and plan on merging the PR, did you see the review I asked a couple days ago?

**Comment by meadowsjared** [comment]
@QarthO  I don't see any review on this PR 🤔 

I checked both my email as well as I took a thorough look at this PR and there are no review comments from you yet. 🤷‍♂️ 

are you sure you `Submit Review` on it?  (until you do, your comments will not be visible to me)

Please make sure you either click `Comment`, `Approve`, or `Request changes`
image
And be sure to click `Submit Review` at the bottom

**Comment by meadowsjared** [comment]
I'm happy to perform changes if you'd prefer any 🥂 
But so far, there are no comments on this PR 😞

**Comment by meadowsjared** [comment]
@QarthO also it says:
image

**Comment by meadowsjared** [comment]
@QarthO when you have time, please make sure to submit your review if you have comments
I'd be happy to look at them but cannot until you do that

**Comment by meadowsjared** [comment]
@QarthO sorry about that, my previous code was missing a crucial step of including the dist folder for the build
that is now corrected! 🤦‍♂️

**Comment by meadowsjared** [comment]
@QarthO 
I looked at build 1.1.3 and noticed that the build was empty. (it doesn't include 

So I looked through your commits and I found what was causing your build to be empty:
https://github.com/QarthO/qHotkeys/commit/97287c60985ad2080bd64f4632946b9d6cdbdcb4#r145225026
It turned out you had removed the "Download build artifacts" step, and as a result, when you publish, it just publishes an empty job. 😞 

If you just re-add these lines: 
```yaml
    - uses: actions/download-artifact@v3
      name: Download build artifacts
      with:
        name: build-artifacts
        path: dist
```
right after the checkout line
https://github.com/QarthO/qHotkeys/blob/886da9a0aca5dd0435719678321f7611c3497797/.github/workflows/ci.yaml#L47
it will make your future builds start working again. 🎉 

You're already uploading them here:
https://github.com/QarthO/qHotkeys/blob/886da9a0aca5dd0435719678321f7611c3497797/.github/workflows/ci.yaml#L34
So that should fix it 🪄

**Comment by QarthO** [comment]
Can you check now?

**Comment by meadowsjared** [comment]
@QarthO 
Yes I can confirm that the npm package is correct now, thank you! 🎉 

In addition to it throwing errors when you try to use it after npm install it, previously if you look in your node_modules\qhotkeys folder, you'd notice it was missing the `dist` folder as shown here:
image
previously that folder was missing from the 1.1.3 build.

However, it is all good now! 🤘 
And in addition, it should be smooth sailing  so any time you update your main branch on github, it will handle all the busy work of publishing now, so you can relax. 🏖️ 🥂 

Thank you for taking care of this, and for adding in the original PR sir! 🚀

**Pull request merged**